visibility/toolboxes/scripts/RangeRingUtils.py

```diff
@@ -0,0 +1,225 @@
+    ''' Make range ring features from a center, and list of distances '''
+    try:
+
+        if sr == "#" or sr == "" or sr == None:
+            sr = srDefault
+
+        rm = RingMaker(centerFC, rangeList, distanceUnits, sr)
+
+        # Create Rings...
+        numCenterPoints = arcpy.GetCount_management(centerFC).getOutput(0)
+        numRingsPerCenter = len(rangeList)
+        totalNumRings = int(numCenterPoints) * int(numRingsPerCenter)
+        totalNumRadials = int(numCenterPoints) * int(numRadials)
+        arcpy.AddMessage("Making rings " + str(totalNumRings) + " (" + str(numRingsPerCenter) + " for " + str(numCenterPoints) + " centers)...")
+        rm.makeRingsFromDistances()
+        outRings = rm.saveRingsAsFeatures(outputRingFeatures)
+
+        # Create Radials...
+        arcpy.AddMessage("Making radials " + str(totalNumRadials) + " (" + str(numRadials) + " for " + str(numCenterPoints) + " centers)...")
+        rm.makeRadials(numRadials)
+        outRadials = rm.saveRadialsAsFeatures(outputRadialFeatures)
+
+        return [outRings, outRadials]
+    except arcpy.ExecuteError:
+        # Get the tool error messages
+        msgs = arcpy.GetMessages()
+        arcpy.AddError(msgs)
+        print(msgs)
+
+    except:
+        # Get the traceback object
+        tb = sys.exc_info()[2]
+        tbinfo = traceback.format_tb(tb)[0]
+
+        # Concatenate information together concerning the error into a message string
+        pymsg = "PYTHON ERRORS:\nTraceback info:\n" + tbinfo + \
+                "\nError Info:\n" + str(sys.exc_info()[1])
+        msgs = "ArcPy ERRORS:\n" + arcpy.GetMessages() + "\n"
+
+        # Return python error messages for use in script tool or Python Window
+        arcpy.AddError(pymsg)
+        arcpy.AddError(msgs)
+        # Print Python error messages for use in Python / Python Window
+        print(pymsg + "\n")
+        print(msgs)
+
+def rangeRingsFromMinMax(centerFC, rangeMin, rangeMax, distanceUnits, numRadials, outputRingFeatures, outputRadialFeatures, sr):
+    ''' Make range ring features from only two distances, a minimum and a maximum '''
+    rangeList = [min(rangeMin, rangeMax), max(rangeMin, rangeMax)]
+    return rangeRingsFromList(centerFC, rangeList, distanceUnits, numRadials, outputRingFeatures, outputRadialFeatures, sr)
+
+def rangeRingsFromInterval(centerFC, numRings, distBetween, distanceUnits, numRadials, outputRingFeatures, outputRadialFeatures, sr):
+    ''' Classic range rings from number of rings, and distance between rings  '''
+    rangeList = [x * distBetween for x in range(1, numRings + 1)]
+    return rangeRingsFromList(centerFC, rangeList, distanceUnits, numRadials, outputRingFeatures, outputRadialFeatures, sr)
+
+
+
+class RingMaker:
+    '''
+    Core class for making range rings.
+    '''
+
+    def __init__(self, center, inputRangeList, distanceUnits, sr):
+        ''' initialize rings '''
+
+        self.deleteme = []
```

Comment:
Yes, everything I see it I think it is another Python Easter egg like: "import this" - http://legacy.python.org/dev/peps/pep-0020/